compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -15671,7 +15674,8 @@ private STNode parseTupleTypeDescOrListConstructorMember(SyntaxKind nextTokenKin
                     return parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_TUPLE);
                 }
 
-                Solution solution = recover(peek(), ParserRuleContext.TUPLE_TYPE_DESC_OR_LIST_CONST_MEMBER);
+                Solution solution = recover(peek(), ParserRuleContext.TUPLE_TYPE_DESC_OR_LIST_CONST_MEMBER,
+                        annots);
```

Comment:
Not related to your PR.. the method-called for re-parse the same rule after inserting a token seems wrong.

Reply:
fixed